### ISSUE TITLE
Enhance Checkpoint Management and Model Serialization

### ISSUE THREAD

**Issue by supremecmd** [kwuking/TimeMixer#142]
BThe current checkpoint management system saves only the best model based on validation performance. This approach limits flexibility for model selection and lacks important metadata that would be useful for resuming training or model analysis.

**Current Limitations:**
- Only the best model checkpoint is saved
- Checkpoint does not include training metadata (epochs, optimizer state, etc.)
- No easy way to resume training from a checkpoint
- No checkpoint pruning strategy for long training runs

**Proposed Solution:**
1. Enhance checkpoint saving to include training metadata:
   - Current epoch/iteration
   - Optimizer state
   - Learning rate scheduler state
   - Training and validation metrics history
   
2. Add checkpoint management features:
   - Option to save periodic checkpoints (e.g., every N epochs)
   - Automatic pruning of old checkpoints to save disk space
   - Option to save both best validation and best training model
   
3. Support for resuming training from saved checkpoints

**Implementation Example:**
```python
def save_checkpoint(self, model, optimizer, scheduler, epoch, metrics, is_best=False, interval=False):
    """Save model checkpoint with training metadata.
    
    Args:
        model: The model to save
        optimizer: The optimizer state
        scheduler: The learning rate scheduler state
        epoch: Current epoch number
        metrics: Dictionary of training/validation metrics
        is_best: Whether this is the best model so far
        interval: Whether this is a periodic interval save
    """
    checkpoint = {
        'epoch': epoch,
        'model_state_dict': model.state_dict(),
        'optimizer_state_dict': optimizer.state_dict(),
        'scheduler_state_dict': scheduler.state_dict() if scheduler else None,
        'metrics': metrics,
        'args': self.args
    }
    
    # Save regular checkpoint
    if interval:
        checkpoint_path = os.path.join(self.checkpoint_dir, f'checkpoint_epoch_{epoch}.pth')
        torch.save(checkpoint, checkpoint_path)
    
    # Save best model
    if is_best:
        best_path = os.path.join(self.checkpoint_dir, 'best_model.pth')
        torch.save(checkpoint, best_path)
        
    # Prune old checkpoints if needed
    if interval:
        self._prune_old_checkpoints(max_keep=5)  # Keep only the 5 most recent checkpoints
```

**Benefits:**
- Ability to resume training from any checkpoint
- Better model selection options (best validation, best training, specific epoch)
- Improved experiment management and analysis
- Disk space management for long training runs
- More robust model saving and loading process

**Comment by kevinliu2000** [comment]
Thanks for the advice.  This management system is not crucial for this. We use any existing tool to achieve it.
